File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # freq_dict=Counter(nums)
        # heap=[]
        # for num,freq in freq_dict.items():
        #     heapq.heappush(heap,(freq,num))
        #     if len(heap)>k:
        #         heapq.heappop(heap)
        # return [item[1] for item in heap]
        #Solution 2: sorting all elements
        freq=Counter(nums) #{1:1,2:2,3:3}
        sorted_items=sorted(freq.items(),key =lambda x:x[1],reverse=True) # sort in descending order of freq 
        # sorted_items:{3:3,2:2,1:1}
        return [item for item,count in sorted_items[:k]]

File: Data Structures & Algorithms/rotate-array/submission-1.java
class Solution {
    public void rotate(int[] nums, int k) {
        
        int n = nums.length;int nk=k%n;
        reverse(nums,0,n-1);
        reverse(nums,0,nk-1);
        reverse(nums,nk,n-1);
    }
    private static int[] reverse(int[] nums, int start, int end){
        while(start<=end){
            int temp=nums[start];
            nums[start]=nums[end];
            nums[end]=temp;
            start++;
            end--;
        }return nums;
}
}